samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -0,0 +1,629 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <zephyr.h>
+#include <gpio.h>
+#include <soc.h>
+#include <assert.h>
+
+#include <settings/settings.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#include <gatt/bas.h>
+#include <bluetooth/services/hids.h>
+#include <bluetooth/services/dis.h>
+
+#define DEVICE_NAME     CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+
+#define BASE_USB_HID_SPEC_VERSION   0x0101
+
+#define OUTPUT_REPORT_MAX_LEN            1
+#define INPUT_REPORT_KEYS_MAX_LEN        8
+#define OUTPUT_REPORT_BIT_MASK_CAPS_LOCK 0x02
+#define INPUT_REP_KEYS_REF_ID            0
```

Comment:
If you add a report id to the usb descriptor, this needs to match the one set there. Note that you cannot then pass this define on to hids_input_rep_send(), as that one start from 0.

Reply:
This seems like a reap problem here.

Reply:
Fixed